packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -144,8 +141,13 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 				});
 
 				// Signal to Vite that we accept HMR updates
-				const SUFFIX = isProduction ? '' : `\nif (import.meta.hot) import.meta.hot.accept((mod) => mod);`;
-
+				let SUFFIX = '';
+				if (isProduction) {
+					SUFFIX += `\nif (import.meta.hot) import.meta.hot.accept((mod) => mod);`;
+				}
```

Comment:
This condition got flipped. We want `import.meta.hot` if `!isProduction`.

Reply:
woof! Great catch, like finding a needle in a haystack